build_cmake/install/install.cmake

```diff
@@ -0,0 +1,68 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+#  See the file LICENSE for redistribution information
+#
+
+cmake_minimum_required(VERSION 3.11.0)
+
+include(GNUInstallDirs)
+
+# Library installs
+
+# Define the public headers for wiredtiger library to be used when installing the target
+set_property(
+    TARGET wiredtiger
+    PROPERTY PUBLIC_HEADER
+    ${CMAKE_BINARY_DIR}/include/wiredtiger.h
```

Comment:
I'm wondering the reason `CMAKE_BINARY_DIR` is used here instead of `CMAKE_SOURCE_DIR`. Guess the 2 variables will have the same value if building from the top level, and could hold different values otherwise.

Reply:
Thanks for bringing this up :) So the values have slightly different meanings!

* `${CMAKE_SOURCE_DIR}` refers to the root project directory (where the top level `CMakeLists.txt` exists)
* `${CMAKE_BINARY_DIR}` refers to the top level _build_ directory. This being the directory you initialized your build. 

e.g For example we can be checked out in `/home/wiredtiger`. When initializing our build we create a directory to perform an out-of-source build (not really allowed to create an in-source build and you generally don't want to pollute your root-level project directory with CMake run-time contents):
```
mkdir build && cd build
cmake ./..
```
In this case:
* `${CMAKE_SOURCE_DIR}` will refer to `/home/wiredtiger`
* `${CMAKE_BINARY_DIR}` will refer to `/home/wiredtiger/build`
We can create as many out-of-source build directories as we desire, all with different names e.g. `build`, `my-better-build`, `debug-stuff` (I usually create a number of different builds when experimenting with different configurations to save the cost of rebuilding and to debug different runs). 

So to answer your original question, in our root CMakeLists we templated the `wiredtiger.h` file with our build configuration ([here](https://github.com/wiredtiger/wiredtiger/blob/wt-7230-cmake-posix-x86/CMakeLists.txt#L74)).  Meaning for the above example the header file will get written to the following location `/home/wiredtiger/build/include/wiredtiger.h`. We ideally want to use `${CMAKE_BINARY_DIR}` to avoid assuming where the configured `wiredtiger.h`will be written to (keeping it portable with different environments).

Reply:
That makes more sense to me now - build portability support is essential. I agree we should avoid in-source builds when possible and apply it as a standard when we integrate CMake into our automated testing system later on.